src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -515,7 +515,7 @@ private DType(DKind kind, string namedValueKind)
             AssertValid();
         }
 
-        internal DType(LazyTypeProvider provider, bool isTable, DisplayNameProvider displayNameProvider = null)
+        internal DType(LazyTypeProvider provider, bool isTable, DisplayNameProvider displayNameProvider = null, HashSet<IExternalTabularDataSource> associatedDataSources = null)
```

Comment:
can we just revert this file 
and LazyTypeProvider.cs too

Reply:
We need this change in conjunction with the ones at line 1072 and 1116 (confirmed to Shantanu at iteration 9)